src/Workspaces/Core/Portable/Workspace/Solution/DocumentState.cs

```diff
@@ -172,14 +186,23 @@ private static string GetSyntaxTreeFilePath(DocumentInfo info)
             return TreeAndVersion.Create(tree, textAndVersion.Version);
         }
 
+        private static int GetIncrementalParseDepth(ValueSource<TreeAndVersion> treeSource)
+        {
+            var incrementalTreeSource = treeSource as IncrementalTreeSource;
+            return incrementalTreeSource != null ? incrementalTreeSource.GetIncrementalParseDepth() : 0;
```

Comment:
I actually prefer Matt's version. I just can't get used to these 'as' expression idioms.